### PR TITLE
coqPackages.elm-extraction: init at 0.1.0

### PR DESCRIPTION
**Pull request by 4ever2**
## Description of changes

Add new coq package https://github.com/AU-COBRA/coq-elm-extraction.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/coq-modules/elm-extraction/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  repo = "coq-elm-extraction";
+  owner = "AU-COBRA";
+  domain = "github.com";
+
+  inherit version;
+  defaultVersion = with versions; switch [coq.coq-version metacoq.version] [
+    { cases = ["8.17" "1.3.1-8.17"]; out = "0.1.0"; }
+    { cases = ["8.18" "1.3.1-8.18"]; out = "0.1.0"; }
+    { cases = ["8.19" "1.3.1-8.19"]; out = "0.1.0"; }
+  ] null;
```
[the remark this answers is not in the document]
Comment (vbgl) [reply pkgs/development/coq-modules/elm-extraction/default.nix:15]: Please note that metacoq has been updated to 1.3.2 in nixpkgs. How do you want to amend the derivation of elm-extraction?

[the remark this answers is not in the document]
Comment (4ever2) [reply pkgs/development/coq-modules/elm-extraction/default.nix]: Thanks for the feedback. I have added MetaCoq 1.3.2 and simplified the case analysis.

**Comment by vbgl** [comment]
Thanks for the update! In order to improve user experience, the attribute path of a coq library in `nixpkgs` matches the logical path of said library in coq (the thing that is written between `From` and `Require`). Here the logical of the library that you are adding is `ElmExtraction`: could you please use that name as `pname`, directory, and attribute name? Thanks.

**Comment by 4ever2** [comment]
https://github.com/NixOS/nixpkgs/pull/341091 should ideally be merged before this PR

**Comment by 4ever2** [comment]
Thanks for the help getting this merged

**Pull request merged**